From: itzexor <itzexor@gmail.com>
Date: Sun, 12 Jun 2016 14:52:21 -0700
Subject: pathbar: avoid gtk+ warnings

If we don't request the slider preferred size in allocation, gtk+
triggers a warning.

The code of the path bar shouldn't do what is doing, and now gtk+ warns
about it. In order to fix it, request the preferred size in size_allocate.

from https://github.com/GNOME/nautilus/commit/b6be37710e235aca9d15af8e74a3d503c0a61ff9
---
 src/nemo-pathbar.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/nemo-pathbar.c b/src/nemo-pathbar.c
index 0d0db13..a6c76cb 100644
--- a/src/nemo-pathbar.c
+++ b/src/nemo-pathbar.c
@@ -618,6 +618,10 @@ nemo_path_bar_size_allocate (GtkWidget     *widget,
     }
     direction = gtk_widget_get_direction (widget);
 
+    gtk_widget_get_preferred_width (path_bar->priv->up_slider_button,
+                                    &path_bar->priv->slider_width,
+                                    NULL);
+
     gtk_widget_get_preferred_size (BUTTON_DATA (path_bar->priv->button_list->data)->button,
                        NULL, &child_requisition);
     width = child_requisition.width;
